src/transformers/models/gptsan_japanese/modeling_gptsan_japanese.py

```diff
@@ -0,0 +1,1028 @@
+        else:
+            pasts = [p.squeeze(1) for p in torch.split(pasts, [1] * self.config.num_layers, 1)]
+            pos = torch.arange(num_input_contexts).to(device) + num_pasts_contexts
+            pos = torch.clip(pos, num_pasts_contexts, self.config.num_contexts - 1)
+
+        ppos = (torch.zeros((self.config.d_model, num_input_contexts)).to(device) + pos).transpose(0, 1).long()
+        hidden += torch.gather(self.position_embeddings.weight, dim=0, index=ppos)
+
+        atten_mask = make_attention_mask_torch(num_input_contexts, num_output_contexts, num_precontext)
+
+        if self.config.num_ext_layers > 0:
+            ete = torch.gather(self.extra_position_embeddings.weight, dim=0, index=ppos)
+        del ppos
+
+        # Prepare head mask if needed
+        if head_mask is not None:
+            head_mask = self.get_head_mask(head_mask, self.config.n_layer)  # n_layer x batch x n_heads x N x N
+
+        out_past_key_values = tuple() if self.config.use_cache or use_cache else None
+        out_hidden_states = (hidden,) if self.config.output_hidden_states or output_hidden_states else None
+        out_attentions = tuple() if self.config.output_attentions or output_attentions else None
+        out_router_logits = tuple() if self.config.output_router_logits or output_router_logits else None
+
+        for layer, past in enumerate(pasts):
+            if layer == self.config.num_switch_layers:
+                hidden = hidden + ete
+
+            output_router_tuple = (
+                self.config.output_router_logits or output_router_logits
+            ) and layer < self.config.num_switch_layers
+            block_output = self.blocks[layer](
+                hidden_states=hidden,
+                layer_past=past,
+                attention_mask=atten_mask,
+                head_mask=head_mask,
+                use_cache=self.config.use_cache or use_cache,
+                output_attentions=self.config.output_attentions or output_attentions,
+                output_router_tuple=output_router_tuple,
+            )
+
+            outpos = 0
+            hidden = block_output[outpos]
+            if self.config.output_hidden_states or output_hidden_states:
+                out_hidden_states += (hidden,)
+            if self.config.use_cache or use_cache:
+                outpos += 1
+                present = block_output[outpos]
+                out_past_key_values += (present,)
+            if self.config.output_attentions or output_attentions:
+                outpos += 1
+                attention_probs = block_output[outpos]
+                out_attentions += (attention_probs,)
+            if output_router_tuple:
+                outpos += 1
+                router_tuple = block_output[outpos]
+                out_router_logits.append(router_tuple[0])
+
+        hidden = self.logits(hidden)
+        hidden = self.logact(hidden)
+
+        logits = torch.einsum("bsc,vc->bsv", hidden, self.embed_tokens.weight)
```

Comment:
If the weights of the model are automatically tied to a `self.lm_head`, you won't have to run this. See [here](https://github.com/ArthurZucker/transformers/blob/7547c82c2f0768cc7e7034e445ce05dc743e6c0c/src/transformers/models/switch_transformers/modeling_switch_transformers.py#L1673), let's follow what is already implemented!